### DISCUSSION TITLE
Automation filter records with Due Date < trigger.timestamp

### DISCUSSION THREAD

**Discussion by magickaito** [Budibase/budibase#6370] [Help]
I want to run a Automation periodically, to query for expired records, and do something about those records.

In the Automation, I try to query for records with due date < a javascript binding.

Where the javascript binding is:

`return new Date($("trigger.timestamp"))`

But that didn't work. I cannot even save this filter. There is some error in the console:
![Screenshot 2022-06-17 at 10 58 48 PM](https://user-images.githubusercontent.com/1195573/174323822-fcaee8eb-5c7e-48da-bab7-27b6cdf48f85.jpg)

I have also tried these bindings, also didn't work:

`{{ trigger.timestamp }}`
`return new Date()'

How should the filter expression be written?

**Comment by melohagan** [comment]
Filter bindings are currently not working in automation queries unfortunately: https://github.com/Budibase/budibase/issues/6090

**Reply by magickaito** [reply]
T_T noted thanks!
